src/Tribe/Linked_Posts.php

```diff
@@ -902,49 +906,110 @@ public function handle_submission_by_post_type( $event_id, $linked_post_type, $s
+			delete_post_meta( $event_id, $linked_post_type_meta_key );
+
+			foreach ( $new_order as $linked_post_id ) {
+				add_post_meta( $event_id, $linked_post_type_meta_key, $linked_post_id );
+			}
+		} else {
+			// We have different Linked Post IDs (adding and/or removing one or more) so possibly need to run through our own methods to trigger those hooks.
+			$posts_to_remove = array_diff( $old_order, $new_order );
 
-			if ( $sorted_priors === $sorted_to_link ) {
-				// If the post IDs are the same (none new nor removed) but not in the same order.
+			foreach ( $posts_to_remove as $key => $unlinked_post_id ) {
+				$this->unlink_post( $event_id, $unlinked_post_id );
+				unset( $temp_old_order[ $key ] );
+			}
 
-				// We do not run our own unlink/link methods because we are not doing that, just re-ordering via `meta_id` by removing all and re-adding in the desired order.
+			// Remove all pre-existing (and non-removed) linked posts to start fresh by re-adding below (for `meta_id` ordering purposes)
+			if ( ! empty( $temp_prior_linked_posts ) ) {
 				delete_post_meta( $event_id, $linked_post_type_meta_key );
+			}
 
-				foreach ( $post_ids_to_link as $linked_post_id ) {
+			foreach ( $new_order as $linked_post_id ) {
+				if ( in_array( $linked_post_id, $old_order ) ) {
+					// Re-add pre-existing ones without our own method because we do not want to trigger those hooks.
 					add_post_meta( $event_id, $linked_post_type_meta_key, $linked_post_id );
+				} else {
+					// Add newly-linked ones via our own method in order to trigger such hooks.
+					$this->link_post( $event_id, $linked_post_id );
 				}
-			} else {
-				// We have different Linked Post IDs (adding and/or removing one or more) so possibly need to run through our own methods to trigger those hooks.
-				$posts_to_remove = array_diff( $prior_linked_posts, $post_ids_to_link );
+			}
+		}
 
-				foreach ( $posts_to_remove as $key => $unlinked_post_id ) {
-					$this->unlink_post( $event_id, $unlinked_post_id );
-					unset( $temp_prior_linked_posts[ $key ] );
-				}
+		return true;
+	}
 
-				// Remove all pre-existing (and non-removed) linked posts to start fresh by re-adding below (for `meta_id` ordering purposes)
-				if ( ! empty( $temp_prior_linked_posts ) ) {
-					delete_post_meta( $event_id, $linked_post_type_meta_key );
-				}
+	/**
+	 * Reorder the meta keys to match the block order.
+	 *
+	 * @since TBD
+	 *
+	 * @param int    $event_id Event ID.
+	 * @param string $linked_post_type The post type of the linked post.
+	 * @param array  $original_order The original IDs/order stored in meta.
+	 *
+	 * @return array
+	 */
+	public function maybe_get_new_order_from_blocks( int $event_id, string $linked_post_type, array $original_order = [] ) {
+		// If the post has blocks, we need to update sorting of the post ids to link so it matches block order.
+		if ( ! has_blocks( $event_id ) ) {
+			return $original_order;
+		}
```

Comment:
Should we also check if someone has blocks editor? Since its possible to have the blocks code in the Content but without the blocks editor active for events?

Reply:
A good question, but in this case we are purely talking about data rather - so even if they are editing via the classic editor, if they have blocks in their post content, we need to observe them.